sdk/state/update.go

```diff
@@ -90,36 +90,63 @@ func (c *Channel) PaymentTxs(p *Payment) (close, decl *txnbuild.Transaction, err
 	return
 }
 
-func (c *Channel) ConfirmPayment(p *Payment) (*Payment, error) {
+func (c *Channel) ConfirmPayment(p *Payment) (payment *Payment, fullySigned bool, err error) {
 	txClose, txDecl, err := c.PaymentTxs(p)
 	if err != nil {
-		return nil, err
+		return p, fullySigned, err
 	}
+
 	// If remote has not signed close, error as is invalid.
-	if err := c.verifySigned(txClose, p.CloseSignatures, c.remoteSigner); err != nil {
-		return nil, fmt.Errorf("incorrect closing transaction, the one given may have different data: %w", err)
+	signed, err := c.verifySigned(txClose, p.CloseSignatures, c.remoteSigner)
+	if err != nil {
+		return p, fullySigned, fmt.Errorf("verifying close signed by remote: %w", err)
+	}
+	if !signed {
+		return p, fullySigned, fmt.Errorf("verifying close signed by remote: not signed by remote")
 	}
+
 	// If local has not signed close, sign.
-	if err := c.verifySigned(txClose, p.CloseSignatures, c.localSigner); err != nil {
-		// TODO - differentiate between wrong signature and missing one
+	signed, err = c.verifySigned(txClose, p.CloseSignatures, c.localSigner)
+	if err != nil {
+		return p, fullySigned, fmt.Errorf("verifying close signed by local: %w", err)
+	}
+	if !signed {
 		txClose, err = txClose.Sign(c.networkPassphrase, c.localSigner)
 		if err != nil {
-			return nil, err
+			return p, fullySigned, fmt.Errorf("signing close with local: %w", err)
 		}
+		p.CloseSignatures = append(p.CloseSignatures, txClose.Signatures()...)
 	}
+
 	// Local should always sign declaration if have not yet.
-	if err := c.verifySigned(txDecl, p.DeclarationSignatures, c.localSigner); err != nil {
+	signed, err = c.verifySigned(txDecl, p.DeclarationSignatures, c.localSigner)
+	if err != nil {
+		return p, fullySigned, fmt.Errorf("verifying declaration signed by local: %w", err)
+	}
+	if !signed {
 		txDecl, err = txDecl.Sign(c.networkPassphrase, c.localSigner)
 		if err != nil {
-			return nil, err
+			return p, fullySigned, err
 		}
+		p.DeclarationSignatures = append(p.DeclarationSignatures, txDecl.Signatures()...)
+	}
+
+	// If remote has not signed close, error as is incomplete.
```

Comment:
```suggestion
	// If remote has not signed declaration, error as is incomplete.
```

Reply:
actually maybe this should say:
`// If remote has not signed declaration, return with not fully signed.`

Reply:
```suggestion
	// If remote has not signed declaration, it is incomplete.
```